lib/xandra.ex

```diff
@@ -420,8 +420,24 @@ defmodule Xandra do
   end
 
   def execute(conn, %Batch{} = batch, options) when is_list(options) do
-    with {:ok, %Error{} = error} <- DBConnection.execute(conn, batch, :no_params, options),
-         do: {:error, error}
+    run(conn, options, fn conn ->
+      case DBConnection.execute(conn, batch, nil, options) do
+        {:ok, %Error{reason: :unprepared}} ->
+          Enum.find_value(batch.queries, fn
+            %Prepared{statement: statement} ->
+              with {:ok, _prepared} <- prepare(conn, statement, Keyword.put(options, :force, true)) do
+                nil
+              end
+            %Simple{} ->
+              nil
+          end) || execute(conn, batch, options)
```

Comment:
This code is really hard to read (at least for me), with the `nil`s used to keep the enumeration going and the `||` after the huge block of code :( It's trying to do too many things at the same time I think.

I think a much simpler solution would instead be something like this

```elixir
case DBConnection.execute(conn, batch, nil, options) do
  {:ok, %Error{reason: :unprepared}} ->
    with :ok <- reprepare_queries_in_batch(conn, batch, options) do
      execute(conn, batch, options)
    end
  {:ok, %Error{} = error} ->
    {:error, error}
  other ->
    other
end)

defp reprepare_queries_in_batch(conn, %Batch{} = batch, options) do
  options = Keyword.put(options, :force, true)
  reprepare_queries_in_batch(conn, batch.queries, options)
end

defp reprepare_queries_in_batch(conn, [%Simple{} | rest], options),
  do: reprepare_queries_in_batch(conn, rest, options)
defp reprepare_queries_in_batch(conn, [%Prepared{statement: statement} | rest], options),
  do: with({:ok, _} <- prepare(conn, statement, options), do: reprepare_queries_in_batch(conn, rest, options)
defp reprepare_queries_in_batch(conn, [], options),
  do: :ok
```

Alternatively, `reprepare_queries_in_batch/3` could be written as:

```elixir
options = Keyword.put(options, :force, true)

Enum.reduce_while(batch.queries, :ok, fn
  %Simple{}, _acc ->
    {:cont, :ok}
  %Prepared{statement: statement}, _acc ->
    case prepare(conn, statement, options) do
      {:ok, _} -> {:cont, :ok}
      {:error, _} = error -> {:halt, error}
    end
end)
```

Thoughts?